testkit-backend/backend.go

```diff
@@ -600,6 +600,12 @@ func (b *backend) handleRequest(req map[string]any) {
 				if executeQueryConfig["txMeta"] != nil {
 					config.TransactionConfigurers = append(config.TransactionConfigurers, neo4j.WithTxMetadata(b.toTxMetadata(executeQueryConfig)))
 				}
+				// Append Auth configuration if it exists
+				if executeQueryConfig["auth"] != nil {
+					auth := executeQueryConfig["auth"].(map[string]any)
+					token := neo4j.BasicAuth(auth["principal"].(string), auth["credentials"].(string), "")
+					config.Auth = &token
+				}
```

Comment:
This can only deserialize basic auth tokens. I think it's be good if TestKit hat the liberty to send any type of auth token (see how those are encoded in the `NewDriver` message). There's already a `getAuth` function in the Go driver's TestKit backend to deserialize different types of auth tokens.

Reply:
Thanks for pointing this out I didn't realise it existed 👍 I've updated the backend to use `getAuth` instead.